help_function/src/onedpl_test_group_load.cpp

```diff
@@ -0,0 +1,223 @@
+  int expected[512];
+  int idx=0;
+  for(int i=0;i< 8 ;++i){
+  for(int j =0;j < sg_sz;++j){
+    for(int k =0;k < 4;++k){ 
+      int val = i * sg_sz * 4 + j * 4 + k;
+      if (idx < 512){
+        expected[idx] = val;
+        idx++;
+         }
+    }
+   }
+  }
+  for (int i = 0; i < 512; ++i) {
+    if (ptr[i] != expected[i]) {
+      std::cout <<" failed\n";
+      std::ostream_iterator<int> Iter(std::cout, ", ");
+      std::copy(ptr, ptr + 512, Iter);
+      std::cout << std::endl;
+      return false;
+    }
+  }
+
+  std::cout <<" pass\n";
+  return true;
+}
+
+template<dpct::group::load_algorithm T>
+bool test_load_blocked_striped_standalone() {
+  // Tests dpct::group::load_algorithm::BLOCK_LOAD_DIRECT & dpct::group::load_algorithm::BLOCK_LOAD_STRIPED 
+  // as standalone methods
+  sycl::queue q;
+  int data[512];
+  for (int i = 0; i < 512; i++) data[i] = i;
+
+  sycl::buffer<int, 1> buffer(data, 512);
+  q.submit([&](sycl::handler &h) {
+    sycl::accessor dacc(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d = dacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          if( T == dpct::group::load_algorithm::BLOCK_LOAD_DIRECT)
+            {dpct::group::load_blocked<4, int>(item, d, thread_data);}
+          else
+            {dpct::group::load_striped<4, int>(item, d, thread_data);}
+          // Write thread_data of each work item at index to the global buffer
+          int global_index = int global_index = item.get_group(2)*item.get_local_range().get(2) + item.get_local_id(2); // Each thread_data has 4 elements
+          #pragma unroll
+            for (int i = 0; i < 4; ++i) {
+                dacc[global_index * 4 + i] = thread_data[i];
+            }
+        });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer, sycl::read_only);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  if(T == dpct::group::load_algorithm::BLOCK_LOAD_DIRECT)
+   {
+     for (int i = 0; i < 512; ++i) {
+      if (ptr[i] != i) {
+        std::cout <<" failed\n";
+        std::ostream_iterator<int> Iter(std::cout, ", ");
+        std::copy(ptr, ptr + 512, Iter);
+        std::cout << std::endl;
+        return false;
+        }
+      }
+    
+      std::cout <<" pass\n";
+      return true;
+  }
+  else{
+    int expected[512];
+    for (int i = 0;i < 128; ++i){
+      for(int j=0;j < 4; ++j){
+        expected[i * 4 +j] = j * 128 +i;  
+      }
+    }
+    for (int i = 0; i < 512; ++i) {
+      if (ptr[i] != expected[i]) {
+        std::cout <<" failed\n";
+        std::ostream_iterator<int> Iter(std::cout, ", ");
+        std::copy(ptr, ptr + 512, Iter);
+        std::cout << std::endl;
+        return false;
+      }
+    }
+  
+    std::cout <<" pass\n";
+    return true;
+     
+  }

```

Comment:
It seems like this could be a reusable helper function which is specialized for the load algorithm and reused for both the standalone and class versions. (not a blocker, but will make for better code)